docs(cn): translate reference/react-dom/server/renderToReadableStream into Chinese (#1228)
reactjs/zh-hans.react.dev: src/content/reference/react-dom/server/renderToReadableStream.md

<Intro>

> `renderToReadableStream` 将 React 树渲染成 [Web 可读流](https://developer.mozilla.org/zh-CN/docs/Web/API/ReadableStream)。

Reviewer: ```suggestion
`renderToReadableStream` 将 React 树渲染后发送至 [Web 可读流](https://developer.mozilla.org/zh-CN/docs/Web/API/ReadableStream)。
```

<Note>

这个 API 依赖 [Web 流](https://developer.mozilla.org/zh-CN/docs/Web/API/Streams_API)，因此在 Node.js 中使用 [`renderToPipeableStream`](/reference/react-dom/server/renderToPipeableStream) 代替。

</Note>

---

## 参考 {/*reference*/}

### `renderToReadableStream(reactNode, options?)` {/*rendertoreadablestream*/}

> 调用 `renderToReadableStream`，将 React 树渲染成 HTML 放到 [Web 可读流](https://developer.mozilla.org/zh-CN/docs/Web/API/ReadableStream) 中。

Reviewer: ```suggestion
调用 `renderToReadableStream`，将 React 树渲染为 HTML 后，发送至 [Web 可读流](https://developer.mozilla.org/zh-CN/docs/Web/API/ReadableStream) 中。
```

  * **可选属性** `progressiveChunkSize`：块中的字节数。[阅读更多默认启发式方法](https://github.com/facebook/react/blob/14c2be8dac2d5482fda8a0906a31d239df8551fc/packages/react-server/src/ReactFizzServer.js#L210-L225)。
  * **可选属性** `signal`：一个 [中止信号](https://developer.mozilla.org/zh-CN/docs/Web/API/AbortSignal)，用于 [中止服务端渲染](#aborting-server-rendering) 并在客户端上渲染其余部分。

#### 返回值 {/*returns*/}

`renderToReadableStream` 返回一个 Promise：

- 如果渲染 [shell](#specifying-what-goes-into-the-shell) 成功，那么 Promise 将 resolve [Web 可读流](https://developer.mozilla.org/zh-CN/docs/Web/API/ReadableStream)。
- 如果渲染 shell 失败，Promise 将 reject。[使用这个输出回滚 shell ](#recovering-from-errors-inside-the-shell)。

返回的流具有附加属性：

* `allReady`：一个 Promise，在所有的渲染完成后 resolve，包括 [shell](#specifying-what-goes-into-the-shell) 和所有额外的 [内容](#streaming-more-content-as-it-loads)。你可以在响应爬虫和静态资源之前执行 `await stream.allReady`，这样就不会得到任何渐进的加载内容，流会包含最终的 HTML。

---

## 用法 {/*usage*/}

> ### 将 React 树按照 HTML 渲染成一个 Web 可读流 {/*rendering-a-react-tree-as-html-to-a-readable-web-stream*/}

Reviewer: ```suggestion
### 将 React 树渲染为 HTML 并发送至 Web 可读流 {/*rendering-a-react-tree-as-html-to-a-readable-web-stream*/}
```

> 调用 `renderToReadableStream` 将 React 渲染成 HTML 的 [Web 可读流](https://developer.mozilla.org/zh-CN/docs/Web/API/ReadableStream)：

Reviewer: ```suggestion
调用 `renderToReadableStream` 将 React 渲染为 HTML 并发送至 [Web 可读流](https://developer.mozilla.org/zh-CN/docs/Web/API/ReadableStream)：
```

hydrateRoot(document, <App />);
```

如此一来，监听事件就挂到了服务器生成的 HTML 上，就可以交互了。

<DeepDive>

#### 从构建输出中读取 CSS 和 JS 的资源路径 {/*reading-css-and-js-asset-paths-from-the-build-output*/}

最终的资源 URL（像 JavaScript 和 CSS 文件）通常在构建后进行散列。例如，你可能最终得到的不是 `styles.css` 而是 `styles.123456.css`。散列静态资源文件名可以保证同一资源的每个不同构建都有不同的文件名。这个方案是有用处的，因为它可以安全地为静态资源开启长期缓存：有了特定名字的文件，其内容就永远不会变。

> 但是，如果构建之后才能知道静态资源的 URL，那就无法将它们放到源代码中。例如，像前面那样将 `"/styles.css"` 硬编码到 JSX 中是行不通的。为了将它们保持在源代码之外，根组件可以从 prop 传递的映射中读取真实文件名：

Reviewer: ```suggestion
但是，如果构建之后才能知道静态资源的 URL，那就无法将它们放到源代码中。例如，像前面那样将 `"/styles.css"` 硬编码到 JSX 中是行不通的。为了将它们保持在源代码之外，根组件可以从 props 传递的映射中读取真实文件名：
```

Author: https://github.com/reactjs/zh-hans.react.dev/issues/2中提到：中文文档中，术语的大小写和单复数请跟下表保持一致。其中props/prop都可以，需与原文一致，此处原文是单数

```

> 在本例中，React 可以更早地开始对页面进行流式传输。只有 `ProfileLayout` 和 `ProfileCover` 是必须先完成渲染的，因为它们没有封装进任何 `<Suspense>` 边界中。但是，如果 `Sidebar` 、 `Friends` 或  `Photos` 需要加载一些数据，React 将发送 HTML 以供 `BigSpinner` 回滚。然后，随着更多的数据变得可用，更多的内容将会被显示，直到所有的内容都可见。

Reviewer: ```suggestion
在本例中，React 可以更早地开始对页面进行流式传输。只有 `ProfileLayout` 和 `ProfileCover` 是必须先完成渲染的，因为它们没有封装进任何 `<Suspense>` 边界中。但是，如果 `Sidebar`、`Friends` 或 `Photos` 需要加载一些数据，React 将发送 HTML 以供 `BigSpinner` 回滚。然后，随着更多的数据变得可用，更多的内容将会被显示，直到所有的内容都可见。
```

在浏览器中，流不需要等待 React 自身的加载，也不需等待应用变得可交互。在任何 `<script>` 标签加载之前，来自服务器中的 HTML 内容将会逐步的显示出来。

[阅读流式 HTML 工作原理的更多信息](https://github.com/reactwg/react-18/discussions/37)。

<Note>

**只有支持 Suspense 的数据源将会激活 Suspense 组件**。它们包括：

- 使用支持 Suspense 的框架获取数据，如 [Relay](https://relay.dev/docs/guided-tour/rendering/loading-states/) 和 [Next.js](https://nextjs.org/docs/getting-started/react-essentials)
- 用 [`lazy`](/reference/react/lazy) 来懒加载组件代码

> Suspense **不会去探测** Effect 内部或事件处理器中获取的数据。

Reviewer: ```suggestion
suspense **不会去探测** Effect 内部或事件处理器中获取的数据。
```

Reviewer: 此处 suspense 更多是作为一个概念，统一使用小写。
